azure-pipelines.yml

```diff
@@ -65,6 +65,32 @@ jobs:
             displayName: "npm build and test"
           - script: npm run test:nolint
             displayName: "npm build and test (no linting)"
+    - job: SmokeTest
+      pool:
+          name: "1ES-Hosted-AzFunc"
+          demands:
+              - ImageOverride -equals MMSUbuntu20.04TLS
+      steps:
+          - task: NodeTool@0
+            inputs:
+                versionSpec: $(NODE_14)
+            displayName: "Install Node.js"
+          - script: npm install
+            displayName: "Install Node.js Dependencies"
+          - script: npm run build
+            displayName: "Build project"
+          - script: npm pack
+            displayName: "Pack npm package"
+          - task: Bash@3
+            displayName: "Create test app"
+            inputs:
+                targetType: "inline"
+                script: "mkdir $(System.DefaultWorkingDirectory)/../test-app && cp -r $(System.DefaultWorkingDirectory)/samples/UnitTesting $(System.DefaultWorkingDirectory)/../test-app"

```

Comment:
Try to use the built in tasks when possible instead of a script. That way it works better on all OS's. There's examples of the npm task above and there's a copy files task with an example here: https://github.com/Azure/azure-functions-nodejs-worker/blob/v3.x/azure-pipelines/build.yml#L76

Ping me offline and I can give you some tips on authoring/testing these yaml files if you want. Azure DevOps has some UI for editing pipelines that frankly isn't that intuitive, but is a lot easier than just modifying the yaml and hoping for the best.

Reply:
Switched to built-in tasks! It took me a while to get it to work on all platforms (see the number of commits :p) but should be good to go now!